activiti-cloud-services-query/activiti-cloud-services-query-graphql-ws/src/main/java/org/activiti/cloud/services/query/qraphql/ws/transport/GraphQLBrokerMessageHandler.java

```diff
@@ -194,21 +219,43 @@ protected void handleMessageInternal(Message<?> message) {
 					}
 			        ExecutionResult executionResult = graphQLSubscriptionExecutor.execute(parameters.getQuery(), parameters.getVariables());
 
-			        if(executionResult.getErrors().isEmpty()) {
-			        	Optional.ofNullable(executionResult.<Publisher<ExecutionResult>>getData())
-			        		.ifPresent(data -> {
-			        			GraphQLBrokerChannelSubscriber subscriber =
-			        					new GraphQLBrokerChannelSubscriber(message, operationPayload.getId(),
-			        							getClientOutboundChannel(),
-			        							bufferTimeSpanMs,
-			        							bufferCount
-			        							);
-
-			        			graphQLsubscriptionRegistry.subscribe(sessionId, operationPayload.getId(), subscriber, () -> {
-			            			data.subscribe(subscriber);
-			        			});
-			        		});
-			        }
+                    if (executionResult.getErrors().isEmpty()) {
```

Comment:
The method `handleMessageInternal` is becoming quite long and complex. What do you thing about extracting some methods from it? You could have methods like `initConnection()`, `startSubscription()`, `stopSubscription()`, `endConnection()`.

Reply:
This is sensible to extract specific handler methods. Done.